### PR TITLE
Token

### PR DESCRIPTION
**Pull request by dmtsui**
Add option to use API token to access conduit. 

This is backwards compatible with cert usage.

### CODE REVIEW

File: canduit.js [hunk 1]
```diff
@@ -90,27 +109,67 @@ Canduit.prototype.exec = function exec (route, params, cb) {
...
+  var qs = params || {};
+  qs['api.token'] = self.token;
+
+  var reqOptions = {
+    url: self.api + route,
+    qs: qs,
+    json: true
+  };
+  var req = request.get(reqOptions,
+      function getOptionsCallback(err, response, data) {
```
Comment (lxe) [inline canduit.js]: Indent should be 2

**Comment by lxe** [comment]
Bring the indent back to 2 as the original. If you wish to move the indent to 4, do that in a separate PR.

File: canduit.js [hunk 2]
```diff
@@ -68,49 +77,80 @@ Canduit.prototype.parseConfigFile = function parseConfigFile (cb) {
...
+
+      cb(null, data.result);
+    });
+  self.logger.info('request made to phabricator', reqOptions);
 };
 
-Canduit.prototype.authenticate = function authenticate (cb) {
-  if (!this.cert) return cb(null, this);
+Canduit.prototype.authenticate = function authenticate(cb) {
+  var self = this;
```
Comment (lxe) [inline canduit.js]: What's the point of reassigning it here? There are no closures with different `this` scope in this function?

Comment (dmtsui) [inline canduit.js]: declaring var self at the top clarifies when self is actually being declared due to hoisting, and using self throughout the function clarifies that the 'this' and 'self' are referring to the same this. I can change it back. This was something suggested to me.

**Comment by lxe** [comment]
Closed in favor of https://github.com/uber/canduit/commit/7c5a30cc287b83f3d891a0042bc519b6e6e7513a

**Pull request closed without merging**